static const double variables in lib/libfrencutils/mosaic_util.c

**Describe the bug**
I get following errors:

```
santha.akella@dlogin06:~/my_stmp/FRE-NCtools/build> make -j 4 |& tee make.log
make  all-recursive
make[1]: Entering directory '/lfs/h2/emc/stmp/santha.akella/FRE-NCtools/build'
Making all in lib
make[2]: Entering directory '/lfs/h2/emc/stmp/santha.akella/FRE-NCtools/build/lib'
Makefile:1118: warning: overriding recipe for target 'clean-local'
Makefile:1113: warning: ignoring old recipe for target 'clean-local'
ftn -DHAVE_CONFIG_H -I. -I../../lib -I..    -module libnfu -real_size 64 -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -O2 -qopenmp -c -o libnfu/nfu.o ../../lib/libnfu/nfu.F90
ftn -DHAVE_CONFIG_H -I. -I../../lib -I..    -module libnfu -real_size 64 -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -O2 -qopenmp -c -o libnfu/nfu.o ../../lib/libnfu/nfu.F90
depbase=`echo libfrencutils/affinity.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../lib -I..    -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.10.6/include -O2 -std=gnu99 -MT libfrencutils/affinity.o -MD -MP -MF $depbase.Tpo -c -o libfrencutils/affinity.o ../../lib/libfrencutils/affinity.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo libfrencutils/create_xgrid.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../lib -I..    -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.10.6/include -O2 -std=gnu99 -MT libfrencutils/create_xgrid.o -MD -MP -MF $depbase.Tpo -c -o libfrencutils/create_xgrid.o ../../lib/libfrencutils/create_xgrid.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo libfrencutils/gradient_c2l.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../lib -I..    -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.10.6/include -O2 -std=gnu99 -MT libfrencutils/gradient_c2l.o -MD -MP -MF $depbase.Tpo -c -o libfrencutils/gradient_c2l.o ../../lib/libfrencutils/gradient_c2l.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo libfrencutils/interp.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../lib -I..    -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.10.6/include -O2 -std=gnu99 -MT libfrencutils/interp.o -MD -MP -MF $depbase.Tpo -c -o libfrencutils/interp.o ../../lib/libfrencutils/interp.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo libfrencutils/mosaic_util.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../lib -I..    -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.10.6/include -O2 -std=gnu99 -MT libfrencutils/mosaic_util.o -MD -MP -MF $depbase.Tpo -c -o libfrencutils/mosaic_util.o ../../lib/libfrencutils/mosaic_util.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo libfrencutils/mpp_domain.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../lib -I..    -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.10.6/include -O2 -std=gnu99 -MT libfrencutils/mpp_domain.o -MD -MP -MF $depbase.Tpo -c -o libfrencutils/mpp_domain.o ../../lib/libfrencutils/mpp_domain.c &&\
mv -f $depbase.Tpo $depbase.Po
depbase=`echo libfrencutils/mpp_io.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I. -I../../lib -I..    -I/apps/prod/hpc-stack/intel-19.1.3.304//netcdf/4.7.4/include -I/apps/prod/hpc-stack/intel-19.1.3.304/hdf5/1.10.6/include -O2 -std=gnu99 -MT libfrencutils/mpp_io.o -MD -MP -MF $depbase.Tpo -c -o libfrencutils/mpp_io.o ../../lib/libfrencutils/mpp_io.c &&\
mv -f $depbase.Tpo $depbase.Po
../../lib/libfrencutils/mosaic_util.c: In function ‘set_the_rotation_matrix’:
../../lib/libfrencutils/mosaic_util.c:1700:36: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                    ^~~
../../lib/libfrencutils/mosaic_util.c:1700:36: note: (near initialization for ‘m[0][0]’)
../../lib/libfrencutils/mosaic_util.c:1700:41: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                         ^~~
../../lib/libfrencutils/mosaic_util.c:1700:41: note: (near initialization for ‘m[0][1]’)
../../lib/libfrencutils/mosaic_util.c:1700:46: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                              ^~~
../../lib/libfrencutils/mosaic_util.c:1700:46: note: (near initialization for ‘m[0][2]’)
../../lib/libfrencutils/mosaic_util.c:1700:53: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                                     ^~~
../../lib/libfrencutils/mosaic_util.c:1700:53: note: (near initialization for ‘m[1][0]’)
../../lib/libfrencutils/mosaic_util.c:1700:58: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                                          ^~~
../../lib/libfrencutils/mosaic_util.c:1700:58: note: (near initialization for ‘m[1][1]’)
../../lib/libfrencutils/mosaic_util.c:1700:63: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                                               ^~~
../../lib/libfrencutils/mosaic_util.c:1700:63: note: (near initialization for ‘m[1][2]’)
../../lib/libfrencutils/mosaic_util.c:1700:69: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                                                     ^~~
../../lib/libfrencutils/mosaic_util.c:1700:69: note: (near initialization for ‘m[2][0]’)
../../lib/libfrencutils/mosaic_util.c:1700:74: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                                                          ^~~
../../lib/libfrencutils/mosaic_util.c:1700:74: note: (near initialization for ‘m[2][1]’)
../../lib/libfrencutils/mosaic_util.c:1700:79: error: initializer element is not constant
   static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
                                                                               ^~~
../../lib/libfrencutils/mosaic_util.c:1700:79: note: (near initialization for ‘m[2][2]’)
make[2]: *** [Makefile:686: libfrencutils/mosaic_util.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/lfs/h2/emc/stmp/santha.akella/FRE-NCtools/build/lib'
make[1]: *** [Makefile:425: all-recursive] Error 1
make[1]: Leaving directory '/lfs/h2/emc/stmp/santha.akella/FRE-NCtools/build'
make: *** [Makefile:366: all] Error 2
```

**To Reproduce**

(On NOAA NCO's WCOSS-2)

```
git clone git@github.com:NOAA-EMC/FRE-NCtools.git

module purge
module load PrgEnv-intel/8.1.0
module load intel/19.1.3.304
module load craype/2.7.17
module load python/3.12.0
module load hdf5/1.10.6
module load netcdf/4.7.4
module load nco/5.2.4
module load cmake/3.20.2

module list

export MPICH_UNEX_BUFFER_SIZE=256m
export MPICH_MAX_SHORT_MSG_SIZE=64000
export MPICH_PTL_UNEX_EVENTS=160k
export KMP_STACKSIZE=2g
export F_UFMTENDIAN=big
export NC_BLKSZ=64K
export CRAY_CPU_TARGET=x86-64

export CC=gcc
export FC=ftn
export CXX=CC

export CFLAGS="-O2 -std=gnu99"
export FCFLAGS="-O2 -qopenmp"

cd FRE-NCtools

autoreconf -i
mkdir build && cd build

../configure --prefix=/lfs/h2/emc/stmp/santha.akella/test2/

make -j 4 |& tee make.log
```


**Expected behavior**
A clean build.

**System Environment**
Describe the system environment, include:
- OS:  x86_64 x86_64 x86_64 GNU/Linux
- Compiler(s): 
- netCDF Version: see ⬇️ 
module load PrgEnv-intel/8.1.0
module load intel/19.1.3.304
module load craype/2.7.17
module load python/3.12.0
module load hdf5/1.10.6
module load netcdf/4.7.4
module load nco/5.2.4
module load cmake/3.20.2


- Configure options: None.
**Additional context**

The fix was: `sed -i 's/static const double m/const double m/g' lib/libfrencutils/mosaic_util.c`

--- a/lib/libfrencutils/mosaic_util.c
+++ b/lib/libfrencutils/mosaic_util.c
@@ -1687,17 +1687,15 @@ int crosses_pole(const double x[] , int n) {
   rotation away from the pole if what is being rotaed is near a pole.
   For rotation matricies formulas and examples, see F.S.Hill, Computer
   Graphics Using OpenGL, @nd ed., Chapter 5.3.
+
+  Note: M_SQRT1_2 is a compile-time constant availble in math.h
 */
 void set_the_rotation_matrix() {
-  static const double is2 = 1.0 /M_SQRT2;
-
-  static const double m00 = 0;
-  static const double m01 = - is2;
-  static const double m02 = is2;
-  static const double m11 = 1.0/2;
-  static const double m12 = 0.5;
-
-  static const double m[3][3] = { {m00, m01, m02}, {m02, m11, m12},{m01, m12, m11} };
+  static const double m[3][3] = {
+      {  0.0,       -M_SQRT1_2, M_SQRT1_2 },
+      {  M_SQRT1_2, 0.5,        0.5       },
+      { -M_SQRT1_2, 0.5,        0.5       }
+  };
 
   for (int i = 0; i < 3; i++) {
     for (int j = 0; j < 3; j++) {

